packages/components/Menu/src/MenuList/MenuList.tsx

```diff
@@ -30,19 +30,22 @@ const MenuStack = stagedComponent((props: React.PropsWithRef<IViewProps> & { gap
 });
 MenuStack.displayName = 'MenuStack';
 
+export const menuListLookup = (layer: string, state: MenuListState, userProps: MenuListProps): boolean => {
+  return state[layer] || userProps[layer] || layer === 'hasMaxHeight' || layer === userProps['minWidth'];
+};
 export const MenuList = compose<MenuListType>({
   displayName: menuListName,
   ...stylingSettings,
   slots: {
     root: MenuStack,
-    ...(Platform.OS === 'macos' && { scrollView: ScrollView }),
+    scrollView: ScrollView,
     ...(Platform.OS === 'macos' && { focusZone: FocusZone }),
   },
   useRender: (userProps: MenuListProps, useSlots: UseSlots<MenuListType>) => {
     const menuList = useMenuList(userProps);
     const menuContext = useMenuContext();
     const menuListContextValue = useMenuListContextValue(menuList);
-    const Slots = useSlots(menuList.props);
+    const Slots = useSlots(menuList.props, (layer) => menuListLookup(layer, menuList, userProps));
```

Comment:
For my own knowledge, why is this filter needed?

Reply:
This looks up the tokens to apply styling to. In this case, `hasMaxHeight` value needs to be read in and forwarded to the appropriate slot.